packages/taquito-michelson-encoder/src/tokens/comparable/timestamp.ts

```diff
@@ -24,14 +24,23 @@ export class TimestampToken extends ComparableToken {
 
   public Encode(args: any[]): any {
     const val = args.pop();
-    return { string: val };
+    if (typeof val === 'number') {
+      return { string: String(val) };
```

Comment:
Should it be { string: String(val) } or { int: String(val) }? Does both work?
Can you add an integration test with a timestamp in the storage to make sure the node accepts it? (maybe 2 tests, one where timestamps is a string and another is a number)

Reply:
wouldn't it be { string: String(val) } in this case since we are parsing from Number to String? I can confirm that String: String(val) works since I tried modifying some of our integration tests to make sure it worked.

Ah sure, I actually had that scenario on the integration test but decided not to put it in since I thought it would be an edge case, but I'll add it in the revision.

Reply:
Or perhaps I'm misunderstanding and we need to retain the original type that was passed instead?

Reply:
update: will be updating it to `int` instead as suggested in this thread: https://tezos-dev.slack.com/archives/C69RKF3PG/p1664818848427549